x-pack/plugins/osquery/public/saved_queries/form/use_saved_query_form.tsx

```diff
@@ -20,11 +22,29 @@ interface UseSavedQueryFormProps {
   handleSubmit: (payload: unknown) => Promise<void>;
 }
 
-export const useSavedQueryForm = ({ defaultValue, handleSubmit }: UseSavedQueryFormProps) =>
-  useForm({
+export const useSavedQueryForm = ({ defaultValue, handleSubmit }: UseSavedQueryFormProps) => {
+  const { data } = useSavedQueries({});
+  const ids: string[] = useMemo<string[]>(
+    () => data?.savedObjects.map((obj) => obj.attributes.id) ?? [],
+    [data]
+  );
+  const idSet = useMemo<Set<string>>(() => {
+    const res = new Set<string>(ids);
+    // @ts-expect-error update types
+    if (defaultValue && defaultValue.id) res.delete(defaultValue.id);
+    return res;
+  }, [ids, defaultValue]);
+  const formSchema = useMemo<ReturnType<typeof createFormSchema>>(() => createFormSchema(idSet), [
+    idSet,
+  ]);
+  return useForm({
     id: SAVED_QUERY_FORM_ID + uuid.v4(),
     schema: formSchema,
-    onSubmit: handleSubmit,
+    onSubmit: async (formData, isValid) => {
```

Comment:
the forms were submitting even when they were invalid before, so i gated them on an `isValid` check. let me know if there's a better place to put this logic.

Reply:
https://github.com/patrykkopycinski/kibana/blob/feat/cases-lens-markdown-plugin/x-pack/plugins/osquery/public/scheduled_query_groups/queries/query_flyout.tsx#L58-L65
the check was done here